src/main/scala/nl.knaw.dans.easy.deposit/docs/DatasetXml.scala

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (C) 2018 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.deposit.docs
+
+import nl.knaw.dans.easy.deposit.docs.DatasetMetadata.DateQualifier.{ DateQualifier, available, created, dateSubmitted }
+import nl.knaw.dans.easy.deposit.docs.DatasetMetadata.{ DateQualifier, _ }
+import nl.knaw.dans.easy.deposit.docs.JsonUtil.InvalidDocumentException
+import org.joda.time.DateTime
+
+import scala.util.Try
+import scala.xml._
+
+object DatasetXml {
+  private val otherDateQualifiers = DateQualifier.values.filterNot { qualifier =>
+    Seq(
+      created, // in ddm:profile
+      available, // in ddm:profile
+      dateSubmitted // generated, ignore if in input
+    ).contains(qualifier)
+  }.toSeq
+
+  // TODO make private together with implicit class RichElem
+  val targetFromQualifier = "qualifier"
+
+  def apply(dm: DatasetMetadata): Try[Elem] = Try {
```

Comment:
I have to do a lot of mental work to grasp and understand what is exactly going on here and what the result will be. This is especially the case with recursive use of methods like `requiredElems` such as in [`orgDetails`](https://github.com/DANS-KNAW/easy-deposit-api/pull/54/files#diff-d0643e927258e9fc26054ff4e75b54c8R105). Given that you have to wrap it in `Some(Seq(...))` feels a bit weird.

I'm sure you have the best intentions with this class, but to me it feels a bit overdesigned. Is there any change we can simplify stuff somewhat more?